packages/govern-core/contracts/OptimisticQueue.sol

```diff
@@ -14,115 +14,157 @@ import "./lib/OptimisticQueueStateLib.sol";
+     */
     function challenge(ERC3000Data.Container memory _container, bytes memory _reason) auth(this.challenge.selector) override public returns (uint256 disputeId) {
         bytes32 containerHash = _container.hash();
-        challengerCache[containerHash] = msg.sender;
+        challengerCache[containerHash] = msg.sender; // cache challenger address while it is needed
         queue[containerHash].checkAndSetState(
             OptimisticQueueStateLib.State.Scheduled,
             OptimisticQueueStateLib.State.Challenged
         );
 
         ERC3000Data.Collateral memory collateral = _container.config.challengeDeposit;
-        collateral.collectFrom(msg.sender);
+        collateral.collectFrom(msg.sender); // pull challenge collateral from sender
 
+        // create dispute on arbitrator
         IArbitrator arbitrator = IArbitrator(_container.config.resolver);
         (address recipient, ERC20 feeToken, uint256 feeAmount) = arbitrator.getDisputeFees();
         require(feeToken.safeTransferFrom(msg.sender, address(this), feeAmount), "queue: bad fee pull");
         require(feeToken.safeApprove(recipient, feeAmount), "queue: bad approve");
-        disputeId = arbitrator.createDispute(2, abi.encode(_container));
+        disputeId = arbitrator.createDispute(2, abi.encode(_container)); // create dispute sending full container ABI encoded (could prob just send payload to save gas)
         require(feeToken.safeApprove(recipient, 0), "queue: bad reset"); // for security with non-compliant tokens (that fail on non-zero to non-zero approvals)
 
+        // submit both arguments as evidence and close evidence period. no more evidence can be submitted and a settlement can't happen (could happen off-protocol)
         emit EvidenceSubmitted(arbitrator, disputeId, _container.payload.submitter, _container.payload.proof, true);
         emit EvidenceSubmitted(arbitrator, disputeId, msg.sender, _reason, true);
         arbitrator.closeEvidencePeriod(disputeId);
 
-        disputeItemCache[arbitrator][disputeId] = containerHash;
+        disputeItemCache[arbitrator][disputeId] = containerHash; // cache a relation between disputeId and containerHash while needed
 
         emit Challenged(containerHash, msg.sender, _reason, disputeId, collateral);
     }
 
-    function resolve(ERC3000Data.Container memory _container, uint256 _resolverId) override public returns (bytes[] memory execResults) {
+    /**
+     * @notice Apply arbitrator's ruling over a challenge once it has come to a final ruling
+     * @param _container A ERC3000Data.Container struct holding both the paylaod being scheduled for execution and
+       the current configuration of the system
+     * @param _disputeId disputeId in the arbitrator in which the dispute over the container was created
+     */
+    function resolve(ERC3000Data.Container memory _container, uint256 _disputeId) override public returns (bytes[] memory execResults) {
```

Comment:
I notice we changed `_disputedId` here but the `IERC3000` is still `resolverId`. Should we change that as well?

Reply:
I am not sure tbh. This is a weird balance